plugin/trino-opa/src/test/java/io/trino/plugin/opa/HttpClientUtils.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.opa;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableListMultimap;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.Response;
+import io.airlift.http.client.StaticBodyGenerator;
+import io.airlift.http.client.testing.TestingHttpClient;
+import io.airlift.http.client.testing.TestingResponse;
+
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Function;
+
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static com.google.common.net.MediaType.JSON_UTF_8;
+import static java.util.Objects.requireNonNull;
+
+public class HttpClientUtils
+{
+    private HttpClientUtils() {}
+
+    public static class RecordingHttpProcessor
```

Comment:
This class isn't thread safe because the setHandler method is not synchronized.  Generally the solution would be to synchronize it. And Also, I'm use `this` as the monitor instead of `this.requests`, but I think this should be change to no have a mutable handle...

Looking at the useage the three unit test classes are all constructing an instance of this class in the `@BeforeEach`, and then each test method is setting the handler.  Since this is a mock http client it is really cheap to construct and does not need to be shared.  So instead of using before/after each at all, I suggest you have the test method create the http client with the actual desired handler.  You can use helper method to create both the HTTP client and the OpaAccessControl.  If necessary you can use try-with-resouces to cleanup, but I'm not sure that is needed.  This would mean that this class could be mostly immutable (except requests), and the tests would be stateless which makes parallelization easier for us.